### PR TITLE
feat(export): asynchronous export service for user list

### PR DESCRIPTION
**Pull request by kschelonka**
An event-driven service for user list export, kicked off by a mutation in list-api. To replace our synchronous export process.

Exports a user's list in parts (10k rows each). These are saved as csvs, which are zipped into an archive once all export processes are completed.

When the export process is complete, the service emits an event to the event bus, which is ingested by our transactional email service. Users will be sent their archive via a signed url, which expires after 14 days. The archive itself expires after 30 days. If a user requests another export in that 30 day period, they wilL be sent a new link to the existing export. When it has expired, a new export may be requested.

[POCKET-10669]

[POCKET-10669]: https://mozilla-hub.atlassian.net/browse/POCKET-10669?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: servers/account-data-deleter/src/queueHandlers/batchDeleteHandler.spec.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
+import { EventEmitter } from 'events';
```
Comment (kschelonka) [inline servers/account-data-deleter/src/queueHandlers/batchDeleteHandler.spec.ts:1]: this file was renamed

File: servers/account-data-deleter/src/queueHandlers/batchDeleteHandler.ts [hunk 2]
```diff
@@ -0,0 +1,174 @@
+import { EventEmitter } from 'events';
```
Comment (kschelonka) [inline servers/account-data-deleter/src/queueHandlers/batchDeleteHandler.ts:1]: The unique logic was pulled out to this class and separated from the abstract class QueueHandler.ts

File: servers/account-data-deleter/src/queueHandlers/queueHandler.ts [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+import { config } from '../config';
+import { SqsMessage } from '../routes/queueDelete';
+import { setTimeout } from 'timers/promises';
+import { SeverityLevel } from '@sentry/types';
+import { unleash } from '../unleash';
+import { type Unleash } from 'unleash-client';
+import { serverLogger } from '@pocket-tools/ts-logger';
+import { QueueConfig } from '../types';
+
+export abstract class QueueHandler {
```
Comment (kschelonka) [inline servers/account-data-deleter/src/queueHandlers/queueHandler.ts:19]: contains shared logic of the two queue handlers (pulling messages, deleting messages if successful, sending event for next poll)

File: servers/account-data-deleter/src/config/index.ts [hunk 4]
```diff
@@ -70,12 +86,13 @@ export const config = {
...
-        process.env.SQS_BATCH_DELETE_QUEUE_URL ||
-        'http://localhost:4566/000000000000/pocket-list-delete-queue',
-    },
+  listExport: {
+    exportBucket:
+      process.env.LIST_EXPORT_PARTS_BUCKET || 'com.getpocket.list-exports',
+    partsPrefix: process.env.LIST_EXPORT_PARTS_PREFIX || '',
+    archivePrefix: process.env.LIST_EXPORT_ARCHIVE_PREFIX || '',
+    queryLimit: 10000,
+    signedUrlExpiry: 60 * 60 * 24 * 7, // 7 days in seconds
```
Comment (bassrock) [inline servers/account-data-deleter/src/config/index.ts:95]: we should prob add 7 days to the email we send so they know

File: infrastructure/account-data-deleter/src/main.ts [hunk 5]
```diff
@@ -60,13 +63,79 @@ class AccountDataDeleter extends TerraformStack {
...
+        filterPolicyScope: 'MessageBody',
+        filterPolicy: JSON.stringify({
+          'detail-type': ['list-export-requested'],
+        }),
+        tags: config.tags,
+      },
+    );
+
+    // Bucket for exports plus auto-expiry rules
+    const exportBucket = new S3Bucket(this, 'list-export-bucket', {
```
[the remark this answers is not in the document]
Comment (kschelonka) [reply infrastructure/account-data-deleter/src/main.ts:95]: that's my assumption but I will check on Monday

File: servers/account-data-deleter/src/config/index.ts (hunk 4 above)
Comment (kschelonka) [reply servers/account-data-deleter/src/config/index.ts:95]: I was thinking that as well

**Review by bassrock — APPROVED** [review]

File: infrastructure/account-data-deleter/src/main.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (kschelonka) [reply infrastructure/account-data-deleter/src/main.ts:95]: confirmed default is private

**Pull request merged**